test/src/org/labkey/test/tests/targetedms/TargetedMSQCConfigureMetricTest.java

```diff
@@ -0,0 +1,94 @@
+package org.labkey.test.tests.targetedms;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.components.targetedms.ConfigureQCMetricsWebPart;
+import org.labkey.test.components.targetedms.QCSummaryWebPart;
+import org.labkey.test.pages.targetedms.PanoramaDashboard;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Category({})
+@BaseWebDriverTest.ClassTimeout(minutes = 2)
+public class TargetedMSQCConfigureMetricTest extends TargetedMSTest
+{
+    private final static String SUBFOLDER_1 = "QC Subfolder 1";
+    private final static String SUBFOLDER_2 = "QC_Subfolder_2";
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        TargetedMSQCConfigureMetricTest init = (TargetedMSQCConfigureMetricTest) getCurrentTest();
+        init.doSetup();
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest)
+    {
+        _containerHelper.deleteProject(getProjectName(), afterTest);
+    }
+
+    private void doSetup()
+    {
+        setupFolder(FolderType.QC);
+    }
+
+   /*
+        Regression coverage for Issue 49427: Inconsistent number of metrics in Panorama QC folder
+    */
+    @Test
+    public void testInconsistentMetricDisplay()
+    {
+        log("Create subfolder, import data and configure custom metrics in " + SUBFOLDER_1);
+        setupSubfolder(getProjectName(), SUBFOLDER_1, FolderType.QC);
+        importData(SProCoP_FILE);
+        navigateToFolder(getProjectName(), SUBFOLDER_1);
+        new PanoramaDashboard(this).getQcSummaryWebPart().clickMenuItem("Configure QC Metrics");
+        ConfigureQCMetricsWebPart configureQCMetrics = new ConfigureQCMetricsWebPart(getDriver());
+        configureQCMetrics.configureMetric("Precursor Area", "Disabled")
+                .configureMetric("Retention Time", "Disabled")
+                .configureMetric("Transition Area", "Disabled")
+                .clickSave();
+
+        log("Create subfolder, import data and configure custom metrics in " + SUBFOLDER_2);
+        goToProjectHome();
+        setupSubfolder(getProjectName(), SUBFOLDER_2, FolderType.QC);
+        importData(SProCoP_FILE);
+        navigateToFolder(getProjectName(), SUBFOLDER_2);
+        new PanoramaDashboard(this).getQcSummaryWebPart().clickMenuItem("Configure QC Metrics");
+        configureQCMetrics = new ConfigureQCMetricsWebPart(getDriver());
+        configureQCMetrics.configureMetric("Precursor Area", "Disabled")
+                .clickSave();
+
+        log("Verify the data is displayed correctly on parent folder");
+        goToProjectHome();
+        QCSummaryWebPart qcSummary = new PanoramaDashboard(this).getQcSummaryWebPart();
+        List<QCSummaryWebPart.QcSummaryTile> subFolderTile = qcSummary.getQcSummaryTiles();
+        if (subFolderTile.get(1).getFolderName().equals(SUBFOLDER_1))
```

Comment:
We should be ordering these folders consistently so I don't think we should need the `if` check here. Were they rendering in different orders for you? If so, I'd like to fix that so it's not random for users.

Reply:
The reason for adding if condition was because we QCSummaryTiles is returning list and order cannot be determined.
The display of tiles on the UI was consistent. 
@labkey-jeckels let me know if you want me to it to Set instead of List.

Reply:
I would expect that the List would consistently reflect the order of the tiles in the UI, so it would be safe to validate/assume which index they're at.